src/client/apiClient.ts

```diff
@@ -0,0 +1,95 @@
+import { RunInformationResponse } from "./responses/runInformationResponse";
+import { SimulationResponse } from "./responses/simulationResponse";
+import { SeriesResponse } from "./responses/seriesResponse";
+import { RequestsSummaryResponse } from "./responses/requestsSummaryResponse";
+import { OutgoingHttpHeaders } from "http";
+
+export interface ApiClient {
+  startSimulation: (simulationId: string, options?: StartSimulationRequest) => Promise<StartSimulationResponse>;
+  getRunInformation: (runId: string) => Promise<RunInformationResponse>;
+  getSimulations: () => Promise<SimulationResponse[]>;
+  abortRun: (runId: string) => Promise<void>;
+  getUserConcurrentMetric: (runId: string, scenario: string) => Promise<SeriesResponse[]>;
+  getRequestsSummary: (runId: string) => Promise<RequestsSummaryResponse>;
+}
+
+export const apiClient = (): ApiClient => {
+  const client = new http.HttpClient();
+  const conf: HttpConf = {
+    // TODO RND-5 configurable
+    baseUrl: "https://cloud.gatling.io/api/public",
+    apiToken: ""
+  };
+  return {
+    startSimulation: (simulationId, options) => {
+      const payload = options ? options : {};
+      return postJson(client, conf, "/simulations/start", payload, { simulation: simulationId });
+    },
+    getRunInformation: (runId) => getJson(client, conf, "/run", { run: runId }),
+    getSimulations: () => getJson(client, conf, "/simulations"),
+    abortRun: (runId) => postJson(client, conf, "/simulations/abort", {}, { run: runId }),
+    getUserConcurrentMetric: (runId, scenario) => getJson(client, conf, "/series", seriesParams(runId, scenario)),
+    getRequestsSummary: (runId) => getJson(client, conf, "/summaries/requests", { run: runId })
+  };
+};
+
+interface HttpConf {
+  baseUrl: string;
+  apiToken: string;
+}
+
+const seriesParams = (runId: string, scenario: string) => ({
+  run: runId,
+  scenario: scenario,
+  group: "",
+  request: "",
+  node: "",
+  remote: "",
+  metric: "usrActive"
+});
+
+const getJson = async <T>(
+  client: http.HttpClient,
+  conf: HttpConf,
+  path: string,
+  params?: Record<string, string>
+): Promise<T> => {
+  const response = await client.getJson<T>(buildUrl(conf, path, params), headers(conf));
+  return handleJsonResponse(response);
+};
+
+const postJson = async <T>(
+  client: http.HttpClient,
+  conf: HttpConf,
+  path: string,
+  payload: any,
+  params?: Record<string, string>
+): Promise<T> => {
+  const response = await client.postJson<T>(buildUrl(conf, path, params), payload, headers(conf));
+  return handleJsonResponse(response);
+};
+
+const headers = (conf: HttpConf): OutgoingHttpHeaders => ({ Authorization: conf.apiToken });
+
+const buildUrl = (conf: HttpConf, path: string, queryParams?: Record<string, string>): string => {
+  const resourceUrl = conf.baseUrl + path;
+  const url = new URL(resourceUrl);
+  if (queryParams) {
+    for (const key in queryParams) {
+      url.searchParams.append(key, queryParams[key]);
+    }
```

Comment:
```suggestion
    Object.entries(queryParams).forEach(([key, value]) => url.searchParams.append(key, value))
```

Reply:
I tend to find the `for` loop slightly more readable/obvious than `forEach` for this kind of imperative code. I can still change it if you insist, I don't care that much. 🤷

(I don't think there's a difference in behavior, right?)

Reply:
There is a small change in performance.
with `entries`, you get both key and value, in comparison with your for, when you have only the key and you have to find the value for each value.


If you prefer the `for` construction:

```suggestion
    for (const [key, value] in Object.entries(queryParams)) {
      url.searchParams.append(key, value);
    }
```

Reply:
Changed to `for ... of Object.entries(...)` 👍